0018-connector-risk-mitigations/0018-connector-risk-mitigations.md

```diff
@@ -16,29 +14,60 @@ The main risk connectors face in Interledger is being unable to fulfill the inco
-* **Denial of Service Protections** - Connectors should take standard [Denial of Service](https://en.bitcoin.it/wiki/Weaknesses#Denial_of_Service_.28DoS.29_attacks) attack prevention steps to prevent attackers from overloading the connectors' servers with network traffic
-* **Prioritizing Fulfillments** - Connectors should prioritize fulfilling transfers over preparing new ones or responding to quote requests and may even have separate processes or machines responsible for those different behaviors
-* **Redundant Instances** - Connectors should run redundant processes and machines to increase the difficulty of interfering with their processing
-* **Avoiding Public Endpoints** - Connectors may reduce attack vectors by using communication channels and protocols that do not entail having a publicly-accessible server (for example using [WebRTC](https://webrtc.org/) or [Virtual Private Networks](https://en.wikipedia.org/wiki/Virtual_private_network) to communicate with peers
+* **Adjusting Transfer Expiry Window** - Connectors determine the window of expiration for a transfer (within a full prepare-fulfill payment cycle) during the prepare phase. The expiration is dependent on the amount of time it takes for a connector to receive a fulfill from from their bilateral party. Forwarding connectors should set the window such that they will be able to receive the fulfillment in time even if the outgoing transfer is executed at the last possible moment.
+* **Denial of Service Protections** - Connectors should take standard [Denial of Service](https://en.bitcoin.it/wiki/Weaknesses#Denial_of_Service_.28DoS.29_attacks) attack prevention steps to prevent attackers from overloading the connectors' servers with network traffic. This includes IP-level protections and ILP-level protections.
+* **Prioritizing Fulfillments** - Connectors should prioritize fulfilling transfers over preparing new ones and may even have separate processes or machines responsible for those different behaviors.
+* **Redundant Instances** - Connectors should run redundant processes and machines to increase the difficulty of interfering with their processing.
+* **Avoiding Public Endpoints** - Connectors may reduce attack vectors by using communication channels and protocols that do not entail having a publicly-accessible server (for example using [WebRTC](https://webrtc.org/) or [Virtual Private Networks](https://en.wikipedia.org/wiki/Virtual_private_network) to communicate with peers.
+
+Note that unconditional payment channels do not require longer timeouts while enabling peers to rebalance frequently and minimize their bilateral trust.
 
-## Liquidity Exhaustion
+## Payments Grieving
```

Comment:
```suggestion
## Liquidity Exhaustion / Payment Griefing
```

The aim of the attack is to exhaust your liquidity, so I think we should call it that. For comparison, a "Denial of Service" is what happens to you, not how it's done

Reply:
I placed a section here because I think you could add a scenario where someone just connects to you, opens channels, and then will take payment but not forward it to their counterparties. Thoughts?

Reply:
That wouldn't be liquidity exhaustion but rather "trolling" in the sense of delaying payments. It could be expanded to a denial of service if you had everyone connect to your malicious nodes and would not forward their payments. In essence, you would waste an on-chain close fee in the case of using payment channels for settlement.

Reply:
Actually, the section is for Payments Griefing, which contains two types of attacks: Liquidity Exhaustion and Counterparty Theft. I think they are separate because liquidity can be exhausted through holding up credit, whereas theft is basically taking a settlement but not passing it on.